api/jobset/v1alpha2/jobset_types.go

```diff
@@ -50,6 +50,8 @@ type JobSetConditionType string
 
 // These are built-in conditions of a JobSet.
 const (
+	// JobSetRunning means the job is running.
+	JobSetRunning JobSetConditionType = "Running"
```

Comment:
I don't want to have a "running" phase because we may not be able to track it accurately and this may cause confusion. Users will think that running means all pods are created, scheduled and actually running, and I don't think this is how we are setting a running phase. 

I suggest to limit the phase to the most deterministic statuses: completed, failed and suspended.

@danielvegamyhre if you have any thoughts.

Reply:
The other suggestion is to not track "Phase" but a "TerminalState", in which case this will have two values: failed and completed (suspended wouldn't be set here because it is not a terminal state).

Reply:
@ahg-g Indeed, we can maintain the TerminalState field and display it only when the status changes to completed or failed, and keep it empty in other cases? Would this be more reasonable?

Reply:
Maybe we should decide between "TerminalState" or "Phase" @ahg-g @danielvegamyhre

Reply:
I would move forward with a TerminalState field that is limited to two values for now: completed and failed

Reply:
And we keep it empty in other cases. right？ @ahg-g

Reply:
yes, it would only be set when the jobset reached a terminal state.

Reply:
> I don't want to have a "running" phase because we may not be able to track it accurately and this may cause confusion. Users will think that running means all pods are created, scheduled and actually running, and I don't think this is how we are setting a running phase.
> 
> I suggest to limit the phase to the most deterministic statuses: completed, failed and suspended.
> 
> @danielvegamyhre if you have any thoughts.

Yes, I think the "running" phase would be tricky to define, easy for users to misinterpret, and difficult to maintain. Sticking with terminal states seems like the right way to go imo, for now at least.